fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+    fetch('/api/todos')
+      .then(res => res.json())
+      .then(data => this.setState({ todos: data }))
+      .catch(err => console.error(err));
+  }
+
+  addTodo(newTodo) {
+    fetch('/api/todos', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newTodo)
+    })
+      .then(res => res.json())
+      .then(this.state.todos.push(newTodo.task));
+    this.getAllTodos();
+  }
+
+  toggleCompleted(todoId) {
+    const newTodos = this.state.todos;
```

Comment:
This will not actually copy the array, make sure to checkout the link in the previous comment. You need to make a copy of the array otherwise you will still mutate the state.

Reply:
This still needs updated